### PR TITLE
Implement f32/f64 comparison operators

### PR DESCRIPTION
**Pull request by axic**
Depends on #448.

Need to be cleaned up and unit tests.

### CODE REVIEW

File: test/unittests/execute_floating_point_test.cpp [hunk 1]
```diff
@@ -66,4 +66,129 @@ TEST(execute_floating_point, f64_add)
...
+        std::nextafter(-Limits::max(), TypeParam{0}),
+        std::nextafter(-TypeParam{1.0}, -Limits::infinity()),
+        -TypeParam{1.0},
+        std::nextafter(-TypeParam{1.0}, TypeParam{0}),
+        std::nextafter(-Limits::min(), -Limits::infinity()),
+        -Limits::min(),
+        std::nextafter(-Limits::min(), TypeParam{0}),
+        std::nextafter(-Limits::denorm_min(), -Limits::infinity()),
+        -Limits::denorm_min(),
+        TypeParam{0},  // FIXME.
```
Comment (axic) [inline test/unittests/execute_floating_point_test.cpp:140]: What is the fixme here?

File: test/unittests/execute_floating_point_test.cpp [hunk 2]
```diff
@@ -66,4 +66,129 @@ TEST(execute_floating_point, f64_add)
...
+                EXPECT_THAT(execute(*inst, ne, {a, b}), Result(uint32_t{i != j})) << a << "!=" << b;
+                EXPECT_THAT(execute(*inst, lt, {a, b}), Result(uint32_t{i < j})) << a << "<" << b;
+                EXPECT_THAT(execute(*inst, gt, {a, b}), Result(uint32_t{i > j})) << a << ">" << b;
+                EXPECT_THAT(execute(*inst, le, {a, b}), Result(uint32_t{i <= j})) << a << "<=" << b;
+                EXPECT_THAT(execute(*inst, ge, {a, b}), Result(uint32_t{i >= j})) << a << ">=" << b;
+            }
+        }
+    }
+
+    // Negative zero.
```
Comment (axic) [inline test/unittests/execute_floating_point_test.cpp:187]: Why are these needed if we have `-0` and `0` in `ordered_values`?

File: test/unittests/execute_floating_point_test.cpp (hunk 1 above)
Comment (chfast) [reply test/unittests/execute_floating_point_test.cpp:140]: It was to add "negative zero" comparison.

File: test/unittests/execute_floating_point_test.cpp (hunk 2 above)
Comment (chfast) [reply test/unittests/execute_floating_point_test.cpp:187]: We don't.

Comment (axic) [reply test/unittests/execute_floating_point_test.cpp:187]: Ah, its -1 and 0. Can we just have -0 there too?

File: lib/fizzy/execute.cpp [hunk 3]
```diff
@@ -1392,33 +1392,81 @@ ExecutionResult execute(Instance& instance, FuncIdx func_idx, span<const Value>
             // effectively no-op
             break;
         }
-        case Instr::f32_add:
+        case Instr::f32_eq:
```
Comment (chfast) [inline lib/fizzy/execute.cpp:1395]: Should be placed after `i64_ge_u`.

File: test/unittests/execute_floating_point_test.cpp (hunk 2 above)
Comment (chfast) [reply test/unittests/execute_floating_point_test.cpp:191]: No.

File: lib/fizzy/execute.cpp (hunk 3 above)
Comment (chfast) [reply lib/fizzy/execute.cpp]: Fixed.

File: test/unittests/execute_floating_point_test.cpp (hunk 2 above)
Comment (axic) [reply test/unittests/execute_floating_point_test.cpp:198]: Why?

Comment (gumb0) [reply test/unittests/execute_floating_point_test.cpp:198]: Why?

Comment (chfast) [reply test/unittests/execute_floating_point_test.cpp:198]: The values in the array must be strictly ordered, otherwise we don't know what result to expect. But `-0 == 0`.

Comment (gumb0) [reply test/unittests/execute_floating_point_test.cpp:198]: I don't see why they need to be strictly ordered, to get the expected result you always calculate it as `i op j` anyway. Also as at some iterations `i < j`, at others `j < i` (and at some `i == j`)

Comment (chfast) [reply test/unittests/execute_floating_point_test.cpp:198]: Please commit a change doing that because I don't see how this suppose to work.

Comment (gumb0) [reply test/unittests/execute_floating_point_test.cpp:199]: Oh I see, I misunderstood how it works first. But I pushed the way I thought it works first, it doesn't rely on strict ordering, what do you think?

File: test/unittests/execute_floating_point_test.cpp [hunk 4]
```diff
@@ -185,12 +186,12 @@ TYPED_TEST(execute_floating_point_types, compare)
             }
             else
             {
-                EXPECT_THAT(execute(*inst, eq, {a, b}), Result(uint32_t{i == j})) << a << "==" << b;
-                EXPECT_THAT(execute(*inst, ne, {a, b}), Result(uint32_t{i != j})) << a << "!=" << b;
-                EXPECT_THAT(execute(*inst, lt, {a, b}), Result(uint32_t{i < j})) << a << "<" << b;
-                EXPECT_THAT(execute(*inst, gt, {a, b}), Result(uint32_t{i > j})) << a << ">" << b;
-                EXPECT_THAT(execute(*inst, le, {a, b}), Result(uint32_t{i <= j})) << a << "<=" << b;
-                EXPECT_THAT(execute(*inst, ge, {a, b}), Result(uint32_t{i >= j})) << a << ">=" << b;
+                EXPECT_THAT(execute(*inst, eq, {a, b}), Result(uint32_t{a == b})) << a << "==" << b;
```
[the remark this answers is not in the document]
Comment (axic) [reply test/unittests/execute_floating_point_test.cpp:189]: That is a good point, I'd rather document these requirements/decision as comments in the test.

[the remark this answers is not in the document]
Comment (gumb0) [reply test/unittests/execute_floating_point_test.cpp:189]: Ok, feel free to delete my commit

[the remark this answers is not in the document]
Comment (chfast) [reply test/unittests/execute_floating_point_test.cpp]: Added comments and dropped the commit.
